shared/src/vm/host_env.rs

```diff
@@ -1282,7 +1282,7 @@ where
     H: StorageHasher,
 {
     let storage = unsafe { env.ctx.storage.get() };
-    let (epoch, gas) = storage.get_block_epoch();
+    let (epoch, gas) = storage.get_current_epoch();
```

Comment:
This wasn't correct. A tx should use the epoch of the current block (the one the tx is being applied in)